src/Monitor/Tracing/Zipkin.hs

```diff
@@ -140,26 +129,55 @@ new (Settings mbHostname mbPort mbEpt mbMgr mbPrd ignore) = liftIO $ do
       , HTTP.port = maybe 9411 fromIntegral mbPort
       , HTTP.requestHeaders = [("content-type", "application/json")]
       }
-  void $ let prd = fromMaybe 0 mbPrd in if prd <= 0
-    then pure Nothing
-    else fmap Just $ forkIO $ forever $ do
-      threadDelay (microSeconds prd)
-      let flush = flushSpans mbEpt tracer req mgr -- Manager is thread-safe.
-      if ignore
-        then catch flush (\(_ :: HttpException) -> pure ())
-        else flush
-  pure $ Zipkin mgr req tracer mbEpt
+    zpk = Zipkin mgr req tracer mbEpt
+  for_ (microSeconds <$> mfilter (> 0) mbPrd) $ \delay -> fork $ forever $ do
```

Comment:
This part is the one I'm the least satisfied with; I took it "as is" from the original PR, but I think it is flawed.
If the `mbPrd` parameter is `Nothing`, the behaviour is messed up: the flush is only performed at the termination of the process (i.e., in most case, when quitting the program) - I suspect (without being 100% sure) because of the intricacies of `mfilter`. Though we typically set this parameter, it would be better not to leave such an aberrant behaviour possible.

Reply:
Yeah, that looks really bad. What is `mbPrd` for? You should not be allowed to not spawn a publishing thread, let's fix this.

Reply:
The documentation for the field reads:
```
 , settingsPublishPeriod :: !(Maybe NominalDiffTime)
  -- ^ If set to a positive value, traces will be flushed in the background every such period.
```

Which, I guess, can be interpreted as "if set to Nothing, there is no regular flush", but it makes very little sense (unless for some very specific tasks like daemons that get restarted or things like this ?). I suggest replacing the Nothing case with a sensible default (every seconds, e.g.).

Reply:
Yeah, let's just rid of the Maybe there.

Reply:
Shall we also remove the `ignore` flag, since it doesn't get used anymore ?